### DISCUSSION TITLE
Unifi Cameras: Motion Sensor & Homekit Integration

### DISCUSSION THREAD

**Discussion by Arosasystems** [koush/scrypted#1394] [General]
Hey everyone, wondering if anyone has figured out a way to configure Unifi cameras' built-in motion sensor to only trigger on Unifi Protect AI events within Homekit? Any tips or insights would be greatly appreciated!

**Comment by fbnksr** [comment]
I'm experiencing the same problem. The motion notification function only operates when set to detect any motion and doesn't function when set to recognize specific types of motion like Person, Animal, or Vehicle.

**Comment by theimo1221** [comment]
Hello @koush,

personally I have anothoer application using https://github.com/hjdhjd/unifi-protect api.
Thus I know and would be capable of implementing an option to restrict the motion sensor to AI-Events like this:
![image](https://github.com/user-attachments/assets/5d0ba7fc-e660-4d6f-ac4a-d8f3b4550ac0)

Would such a PR be appreciated? If so I'll happily implement it 😄 

Best Regards

Thiemo

**Comment by koush** [comment]
no, this would not be accepted. this would gate motion events, which trigger apple detections, behind unifi's object detection. that means apple would never detect anything unless unifi detected it first, resulting in missed detections.

**Reply by theimo1221** [reply]
Thanks for the fast reply 👍 

Whilst I totally agree with your point of potential missing detections, in my opinion an option (with default being disabled --> no filtering) would give this choice to the user (maybe even on a per camera basis?)
E.g. in my use case, I like/use notifications on the apple tv for convenience, but having it on any motion, results in e.g. car-beams of drive-by vehicels triggering motion forcing me to deactivate it altogether....

If this would be okay for you, I'd start working on a PR implementing this 😄

**Reply by koush** [reply]
you can fork it and publish your own version, but I think this is an incorrect implementation that will silently result in missed detections. there are several ways to make a motion sensor trigger from external sources that could be used. having to hack it into the plugin is not necessary.

**Reply by theimo1221** [reply]
The point of it being kinda hacky is true, but in some scenarios I'm not sure if there is any other solution.

I might be missing or misunderstanding something here: Within unifi I can configure specific smart zones and devices like AI-Pro or even the new G6 have onboard AI-Detection, thus making the results quite good respecting the users setting.

On Apple TV I have these options:
<img width="1034" alt="image" src="https://github.com/user-attachments/assets/4c08ba44-9798-4f66-a5d7-be8aa44d060e" />
As far as I understand correctly I have two options to reach the goal of notification on detection: 
1. Alert on motion --> Triggers to often (e.g. car beams)
2. Alert on specific motion --> Triggers "different" (some events are too much [outside smart zone], some missed by apple)

Yes, I can potentially trigger additional sensors, but they would be ignored by apple tv notification handling.
